src/main/java/org/simulator/examples/SBMLTestSuiteRunnerWrapper.java

```diff
@@ -0,0 +1,180 @@
+import org.simulator.comp.CompSimulator;
+import org.simulator.io.CSVImporter;
+import org.simulator.math.odes.AbstractDESSolver;
+import org.simulator.math.odes.MultiTable;
+import org.simulator.math.odes.RosenbrockSolver;
+import org.simulator.sbml.SBMLinterpreter;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.*;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * <p>A wrapper class used to verify tests in the SBML Test Runner</p>
+ *
+ * <p>
+ * The results of the simulation by this class are stored as a CSV file in the specified output
+ * directory in SBML Test Runner. Then the SBML Test Runner creates the distance plots comparing
+ * this results with the pre-defined results from SBML Test Suite under the criteria given at this
+ * <a href="http://sbml.org/Software/SBML_Test_Suite/Case_Descriptions#The_.22settings.22_file" target="_blank">link</a>
+ * </p>
+ */
+public class SBMLTestSuiteRunnerWrapper {
+
+    /**
+     * Runs a simulation of a SBML file and writes result to a specified CSV file
+     * @param args
+     * @throws IOException
+     * @throws XMLStreamException
+     * @throws ModelOverdeterminedException
+     * @throws DerivativeException
+     */
+    public static void main(String[] args) throws IOException, XMLStreamException, ModelOverdeterminedException, DerivativeException {
+
+        // Configuration
+        String dirPath = args[0];
+        String currentCase = args[1];
+        String outputDirPath = args[2];
+        String level = args[3];
+        String  version = args[4];
+
+        String filePath = dirPath + '/' + currentCase + '/' + currentCase + "-sbml-l" + level + 'v' + version + ".xml";
+        String settingsPath = dirPath + '/' + currentCase + '/' + currentCase + "-settings.txt";
+        String outputFilePath = outputDirPath + '/' + currentCase + ".csv";
+        String resultsPath = dirPath + '/' + currentCase + '/' + currentCase + "-results.csv";
+
+        Properties properties = new Properties();
+        properties.load(new BufferedReader(new FileReader(settingsPath)));
+        double duration;
+        double steps = (!properties.getProperty("steps").isEmpty()) ? Double.parseDouble(properties.getProperty("steps")) : 0d;
+        Map<String, Boolean> amountHash = new HashMap<String, Boolean>();
+        String[] amounts = String.valueOf(properties.getProperty("amount")).split(",");
+        String[] concentrations = String.valueOf(
+                properties.getProperty("concentration")).split(",");
+
+        for (String s : amounts) {
+            s = s.trim();
+            if (!s.isEmpty()) {
+                amountHash.put(s, true);
+            }
+        }
+
+        for (String s : concentrations) {
+            s = s.trim();
+            if (!s.isEmpty()) {
+                amountHash.put(s, false);
+            }
+        }
```

Comment:
You putting the concentrations in the amountHash? Is this correct?

Reply:
amountHash keeps track of the following:
- Is true for those variables whose output in results file is in amount units
- Is false for those variables whose output in results file is in concentration units

It's a String to Boolean hashmap.